java/org/contikios/cooja/Cooja.java

```diff
@@ -863,6 +863,49 @@ public Plugin getPlugin(String classname) {
     return null;
   }
 
+  /**
+   * Returns a started plugin that is assignable by the given class, if any.
+   *
+   * @param type a class specifying the type of plugin
+   * @return Plugin instance
+   */
+  public <T extends Plugin> T getPlugin(Class<T> type) {
+    for (Plugin p: startedPlugins) {
+      if (type.isInstance(p)) {
+        return type.cast(p);
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Returns all started plugins that are assignable by the given class, if any.
+   *
+   * @param type a class specifying the type of plugin
+   * @return A list of plugin instances
+   */
+  public <T extends Plugin> List<T> getAllPlugins(Class<T> type) {
```

Comment:
I think the "All" part of the names can be dropped, getPlugin returns a single plugin, getPlugins returns multiple plugins.

Reply:
Agreed. Updated.